src/v/cluster/health_monitor_frontend.cc

```diff
@@ -63,29 +72,15 @@ health_monitor_frontend::collect_node_health(node_report_filter f) {
           return be.collect_current_node_health(std::move(f));
       });
 }
-
-// Return status of single node
-ss::future<result<std::vector<node_state>>>
-health_monitor_frontend::get_nodes_status(
-  model::timeout_clock::time_point deadline) {
-    return dispatch_to_backend([deadline](health_monitor_backend& be) {
-        // build filter
-        cluster_report_filter filter{
-          .node_report_filter = node_report_filter{
-            .include_partitions = include_partitions_info::no,
-          }};
-        return be.get_cluster_health(filter, force_refresh::no, deadline)
-          .then([](result<cluster_health_report> res) {
-              using ret_t = result<std::vector<node_state>>;
-              if (!res) {
-                  return ret_t(res.error());
-              }
-
-              return ret_t(std::move(res.value().node_states));
-          });
-    });
+std::optional<alive>
+health_monitor_frontend::is_alive(model::node_id id) const {
+    auto status = _node_status_table.local().get_node_status(id);
+    if (!status) {
+        return std::nullopt;
+    }
+    return alive(
+      status->last_seen >= model::timeout_clock::now() - _alive_timeout());
```

Comment:
nit: I think it is better to rewrite it with addition instead of subtraction to avoid overflow. Who knows what's in this timeout_clock::now().